soc/nxp/imxrt/imxrt118x/soc.c

```diff
@@ -406,18 +454,28 @@ static int imxrt_init(void)
 	XCACHE_EnableCache(XCACHE_PS);
 #elif defined(CONFIG_SOC_MIMXRT1189_CM7)
 	sys_cache_data_enable();
+#endif
+
+	/*
+	 * TODO: cache enablement should be with proper MPU configuration.
+	 * Otherwise it causes fault when accessing NETC registers.
+	 * Also NonCacheable section should be supported since HAL driver requires.
+	 * So, disabling cache here is a workaround to use NETC.
+	 */
+#if defined(CONFIG_ETH_NXP_IMX_NETC) && (DT_CHILD_NUM_STATUS_OKAY(DT_NODELABEL(netc)) != 0)
+#if defined(CONFIG_SOC_MIMXRT1189_CM33)
+	XCACHE_DisableCache(XCACHE_PS);
+	XCACHE_DisableCache(XCACHE_PC);
```

Comment:
Have you looked into fixing this.

Reply:
No, actually. I am not familar with RT1180 platform.
I reported the problem to board owner Lucien Zhao, and requested to handle this if he gets any chance.
To enable cache, at least MPU configuration and noncache section used in NXP HAL should be handled properly.
This should be another topic. 
If we accept the workaround, we can use it for now, and do not let it block any other features merging.

Thanks.